### DISCUSSION TITLE
lxqt-panel delayed before displaying - Debian 10 and 11

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2082] [Q&A] [answered]
With both Debian 10 (buster) and now Debian 11 (bullseye) (amd64), when LXQt is loading in, the appearance/loading of lxqt-panel is delayed for close to 20 seconds, before it appears. 

Would this be an issue with Debian (since this occurs with two versions now) or with lxqt-panel itself? I have LXQt installed on two computers and this occurs on both.

Thanks in advance.

**Comment by stefonarch** [comment, marked as answer]
I was using buster and now bullseye on my laptop, and did never see any delay. 
So I think the reason is somewhere else. Autostart apps with "wait for systray? WIndow manager/compositor?

**Reply by edwardp3** [reply]
This is the current Autostart window:

![snapshot80](https://user-images.githubusercontent.com/73129450/135640543-a9c18689-130f-4f97-95e4-02bf1dbe5ed6.png)

As to the Window Manager, it's using Breeze as the default and I'm not seeing anything on those tabs pertaining to the panel.

**Reply by tsujan** [reply]
Elsewhere, there was an old discussion (2 years ago -- can't find the link). The delay only happened in a virtual machine and seemed to be caused by xfwm4 somehow, although I tried xfwm4 on Manjaro + VM and saw no delay.

If you use xfwm4, make sure its settings are compatible with your graphic card. Also, make sure that its shortcuts aren't in conflict with LXQt's.

> As to the Window Manager, it's using Breeze

Breeze isn't a window manager.

**Reply by tsujan** [reply]
Also, please note that Bullseye's LXQt is outdated.

**Reply by stefonarch** [reply]
_Breeze_ is a widget theme, your window manager (title bar ecc) looks like xfwm4 from the screenshot.
Anyway I cannot reproduce this also with xfwm4.
Maybe you can see something relevant in  `cat ~.xsession-errors | grep lxqt`

**Reply by edwardp3** [reply]
xfwm4 and xfwm4-theme-breeze are installed, but I am not using xfwm4 as the window manager, it's using tdm-trinity (Trinity Desktop's window manager). I noticed 0.16 came with Bullseye. Given when 0.17 was released, I think Debian should have included that version with Bullseye.

The video card:

`01:05.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] RS780L [Radeon 3000] [1002:9616]`

**Reply by edwardp3** [reply]
```
~$ cat ~.xsession-errors | grep lxqt
cat: '~.xsession-errors': No such file or directory
```

Under LXQT Settings/Session Settings, it is showing xfwm4 as the window manager. I must be confusing the window manager with the login manager (tde-trinity). When I select Search on that screen, it defaults to /usr/bin to select a different window manager, but none are listed. I know LXDE uses openbox, would this be an acceptable alternative to xfwm4?

**Reply by tsujan** [reply]
> it's using tdm-trinity

That isn't a window manager. A window manager is an app that draws the titlebars of your windows, controls how windows appear, probably provides compositing, etc. Your screenshot shows you're using xfwm4.

**Reply by tsujan** [reply]
> I know LXDE uses openbox, would this be an acceptable alternative to xfwm4?

Yes, it is but it lacks compositing (no shadows, no translucency). You could try it to know if it removes the delay. If it does, you could switch back to xfwm4 and find which settings of it might have caused the delay.

**Reply by edwardp3** [reply]
Just installed openbox (with required dependencies) and after a reboot, the panel came up immediately.

**Reply by tsujan** [reply]
> Just installed openbox (with required dependencies) and after a reboot, the panel came up immediately.

Good. Then the problem is related to xfwm4 or its version that Debian has.

Since the problem can't be reproduced by most users -- even under Debian -- I think it's about some xfwm4's settings. Only you could find it.

**Reply by stefonarch** [reply]
> cat: '~.xsession-errors': No such file or directory

tipo, sorry: ` cat ~/.xsession-errors`

For translucency and shadows you can install  compton (which is deprecated but has a GUI compton-conf) or picom (which takes the settings from compton).

**Reply by edwardp3** [reply]
> > cat: '~.xsession-errors': No such file or directory
> 
> tipo, sorry: ` cat ~/.xsession-errors`
> 
> For translucency and shadows you can install compton (which is deprecated but has a GUI compton-conf) or picom (which takes the settings from compton).

```
$ cat ~/.xsession-errors | grep lxqt
dbus-update-activation-environment: setting DESKTOP_SESSION=lxqt
dbus-update-activation-environment: setting XDG_SESSION_DESKTOP=lxqt
```
picom is also on GitHub, but there seems to be an issue with it pertaining to high CPU usage. One poster reported going back to a previous version which corrected their issue.

**Reply by edwardp3** [reply]
After switching the window manager from openbox back to xfwm4, lxqt-panel has been appearing much faster after login. No changes were made to any of the settings. 

Debian is providing version 4.16.1 of xfwm4 with bullseye.
